docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,375 @@
+
+- `reference`: full reference by digest of the referrer file
+- `mediaType`: media type of the referrer
+- `size`: referrer file size in bytes
+- `digest`: digest of the attached referrer file
+- `artifactType`: artifact type of the referrer
+- `annotations`: contains arbitrary metadata in a referrer
+
+For example, attach two files to an image and show the descriptor metadata of the referrer in JSON format.
+
+```bash
+oras attach $REGISTRY/$REPO:$TAG --artifact-type example/report-and-sbom vul-report.json:example/vul-scan sbom.spdx:example/sbom --format json
+```
+
+```json
+{
+  "reference": "$REGISTRY/$REPO@sha256:0afd0f0c35f98dcb607de0051be7ebefd942eef1e3a6d26eefd1b2d80f2affbe",
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "digest": "sha256:0afd0f0c35f98dcb607de0051be7ebefd942eef1e3a6d26eefd1b2d80f2affbe",
+  "size": 923,
+  "annotations": {
+    "org.opencontainers.image.created": "2023-12-15T08:59:21Z"
+  },
+  "artifactType": "example/report-and-sbom"
+}
+```
+
+### oras discover
+
+View an artifact's referrers. The default output should be listed in a tree view.
+
+```bash
+oras discover $REGISTRY/$REPO:$TAG --format tree
+```
+
+```console
+$REGISTRY/$REPO@sha256:5cb894d0c94c56894e160ad2eeb19a123b4d2155374e7709f43f8c0c2f249fe2
+├── application/vnd.oci.empty.v1+json
+│   └── sha256:0db683b656132cede5360f42bc52541f3386b30ce685e6e63ff93ced54423fb8
+└── application/vnd.cncf.notary.signature
+    └── sha256:476d43120d3799fa76d7706f741beca73f5ff4149c8b6db3bd516a73d4a82fc1
+```
+
+View an artifact's referrers manifest in pretty JSON output. The following fields should be outputted:
+
+- `manifests`: the list of referrers
+  - `reference`: full reference by digest of the referrer
+  - `mediaType`: media type of the referrer
+  - `size`: referrer file size in bytes
+  - `digest`: digest of the referrer
+  - `artifactType`: artifact type of a referrer
+  - `annotations`: contains arbitrary metadata in a referrer
+
+See an example:
+
+```bash
+oras discover $REGISTRY/$REPO:v1 --format json
+```
+
+```json
+{
+  "manifests": [
+    {
+      "reference": "$REGISTRY/$REPO@sha256:0db683b656132cede5360f42bc52541f3386b30ce685e6e63ff93ced54423fb8"
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:0db683b656132cede5360f42bc52541f3386b30ce685e6e63ff93ced54423fb8",
+      "size": 964,
+      "annotations": {
+        "org.opencontainers.image.created": "2023-12-14T13:48:32Z"
+      },
+      "artifactType": "application/vnd.oci.empty.v1+json"
+    }
+  ]
+}
+```
+
+> [!NOTE]
+> The `--format` flag will replace the existing `--output` flag. The `--output` will be marked as "deprecated" in ORAS v1.2.0 and will be removed in the future releases. 
+
+## Feature status
+
+This feature is marked as experimental in its first iteration since it is still in development and is available in a part of ORAS CLI commands.
```

Comment:
Should mention which feature is recommended:
- Recommended (documented and will break): `--format <type>[=<parameters>]`
- Usable (undocumented, easier to use): `--format <type> [--template <parameters>]`
Also, according to our last offline discussion, both `--format` and `--template` are marked as experimental. In later versions, `--format` can be upgraded to `preview`, but `--template` still remain `experimental` or removed.

Reply:
Added this note in the feature status section